tasks/src/main/scala/dagr/tasks/ScatterGather.scala

```diff
@@ -93,14 +102,21 @@ object ScatterGather {
   private class PartitionerWrapper[A](partitioner: Partitioner[A]) extends Scatter[A] {
     override def getTasks: Traversable[_ <: Task] = Some(partitioner)
 
-    override def gather[B <: Task](f: (Seq[A]) => B): Gather[A,B] =
+    override def gather[B <: Task](f: Seq[A] => B): Gather[A,B] =
       throw new UnsupportedOperationException("gather not supported on an unmapped Scatter")
 
     override def map[T <: Task](f: A => T): Scatter[T] = {
       val sub = new PrimarySubScatter(partitioner, f)
       this ==> sub
       sub
     }
+
+
+    override def groupBy[K, B <: Task](f: A => K, g: (K, Seq[A]) => B) : Scatter[B] = {
```

Comment:
I think this could probably just be another `throw new UnsupportedOperationException` because it seems very odd to me that you would scatter, do no mapping, and then immediately try and group.  If that's the case, why not just scatter on the right thing to begin with?